src/VirtoCommerce.SitemapsModule.Web/Controllers/Api/SitemapsModuleApiController.cs

```diff
@@ -341,36 +341,37 @@ void SendNotificationWithProgressInfo(ExportImportProgressInfo c)
                 var relativeUrl = $"tmp/sitemap-{storeId}.zip";
                 var localTmpFolder = _hostingEnvironment.MapPath(Path.Combine("~/", _platformOptions.LocalUploadFolderPath, "tmp"));
                 var localTmpPath = Path.Combine(localTmpFolder, $"sitemap-{storeId}.zip");
+
+                // Create directory if not exist
                 if (!Directory.Exists(localTmpFolder))
-                {
                     Directory.CreateDirectory(localTmpFolder);
-                }
 
-                //Import first to local tmp folder because Azure blob storage doesn't support some special file access mode 
-                using (var stream = SystemFile.Open(localTmpPath, FileMode.OpenOrCreate))
+                // Remove old file if exist
+                if (SystemFile.Exists(localTmpPath))
+                    SystemFile.Delete(localTmpPath);
```

Comment:
Probably deletion should be done only after successful creation operation, otherwise we could delete old sitemaps and do not create the broken one - so everything will be lost.